spec/controllers/api/v1/punches_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe Api::V1::PunchesController, :type => :controller do
+
+  let!(:user) { create(:user) }
+  let!(:project) { create(:project, company: user.company) }
+  let!(:company) { user.company }
+  let!(:punches) { Punch.create!(created_at: '2022-04-26T09:53:51.248-03:00', from: '2021-10-26T08:00:00.000-03:00', to: '2021-10-26T12:00:00.000-03:00', extra_hour: false, project_id: project.id, company_id: company.id, user_id: user.id) }
```

Comment:
This line seems a little long, you could perhaps split this into more lines.

Reply:
Don't we have a factory for the Punch model in order to use `create(:punch)`?